ion/integration_test.go

```diff
@@ -673,7 +705,7 @@ func readCurrentValue(t *testing.T, reader Reader) ionItem {
 	currentType := reader.Type()
 	// Excluding SymbolType from setting value to 'null.symbol' because readers can read "'null.symbol'" as text: 'null.symbol'.
 	if reader.IsNull() && reader.Type() != SymbolType {
```

Comment:
we can remove `reader.Type() != SymbolType` and add 

```
			if val == nil {
				err := writer.WriteNullType(SymbolType)
				if err != nil {
					t.Fatal(err)
				}
```

under `case SymbolType`

Reply:
Do we even need this `reader.IsNull()` conditional? if a value is nil, the value APIs will return a `nil`

Reply:
At this point we still have not read the value. This is to avoid going through the switch statement; as soon as we get to a null value we create corresponding ```ionItem``` for it.

Can you explain the code snippet please? Why should it be ```WriteNullType(SymbolType)```, what if we are reading ```null.timestamp```?

Reply:
> At this point we still have not read the value. This is to avoid going through the switch statement; as soon as we get to a null value we create corresponding ionItem for it.

I think it would be good idea if we let it go through the switch statement, to test that the value APIs do return `nil` instead of the test setting the value to `nil`

> Can you explain the code snippet please? Why should it be WriteNullType(SymbolType), what if we are reading null.timestamp?

My bad, I thought the reason why we were excluding symbolType being set to `nil` was because the writer wasn't writing a `null` symbol correctly. Disregard the comment

Reply:
Done